devenv/main.tf

```diff
@@ -7,10 +7,10 @@ provider "azurerm" {
 }
 
 provider "kubernetes" {
-  host                   = azurerm_kubernetes_cluster.cluster.kube_config.0.host
-  client_certificate     = base64decode(azurerm_kubernetes_cluster.cluster.kube_config.0.client_certificate)
-  client_key             = base64decode(azurerm_kubernetes_cluster.cluster.kube_config.0.client_key)
-  cluster_ca_certificate = base64decode(azurerm_kubernetes_cluster.cluster.kube_config.0.cluster_ca_certificate)
+  host                   = var.private-dns ? azurerm_kubernetes_cluster.cluster-private[0].kube_config.0.host : azurerm_kubernetes_cluster.cluster[0].kube_config.0.host
```

Comment:
With more additions to config and complex feature I think this wont be viable. I think it would be better to have separation in the tf files used for specific tests/dev than clump it all into one place

Reply:
with the number of repeated resources across both test types I wasn't sure, but are you basically saying we should have one TF file for private clusters and one for public?